### PR TITLE
[TASK] Streamline CI-Setup, add tests and Composer 2 support for 1.5.x

### PR DESCRIPTION
**Pull request by benjaminkott**

### CODE REVIEW

File: src/Plugin/Core/InstallerScripts/AutoloadConnector.php [hunk 1]
```diff
@@ -53,12 +57,19 @@ public function run(Event $event): bool
 
         $composerConfig = $composer->getConfig();
         $localRepository = $composer->getRepositoryManager()->getLocalRepository();
-        $package = $localRepository->findPackage('typo3/cms', new EmptyConstraint());
+        $package = $localRepository->findPackage('typo3/cms', class_exists(EmptyConstraint::class) ? new EmptyConstraint() : new MatchAllConstraint());
```
Comment (helhum) [inline src/Plugin/Core/InstallerScripts/AutoloadConnector.php:60]: can be simplified to $localRepository->findPackage('typo3/cms', '*');

File: src/Plugin/Core/InstallerScripts/AutoloadConnector.php [hunk 2]
```diff
@@ -53,12 +57,19 @@ public function run(Event $event): bool
...
 
         $defaultVendorDir = \Composer\Config::$defaultConfig['vendor-dir'];
 
         $packagePath = $composer->getInstallationManager()->getInstallPath($package);
-        $jsonFile = new \Composer\Json\JsonFile($packagePath . DIRECTORY_SEPARATOR . 'composer.json', new \Composer\Util\RemoteFilesystem($io));
```
Comment (helhum) [inline src/Plugin/Core/InstallerScripts/AutoloadConnector.php:60]: no remote filesystem here as well

File: src/Plugin/Core/InstallerScripts/WebDirectory.php [hunk 3]
```diff
@@ -91,7 +92,7 @@ private function initializeSymlinks()
         $webDir = $this->filesystem->normalizePath($this->pluginConfig->get('web-dir'));
         $this->filesystem->ensureDirectoryExists($webDir);
         $localRepository = $this->composer->getRepositoryManager()->getLocalRepository();
-        $package = $localRepository->findPackage('typo3/cms', new EmptyConstraint());
+        $package = $localRepository->findPackage('typo3/cms', class_exists(EmptyConstraint::class) ? new EmptyConstraint() : new MatchAllConstraint());
```
Comment (helhum) [inline src/Plugin/Core/InstallerScripts/WebDirectory.php:95]: `'*'`

File: src/Plugin/Core/ScriptDispatcher.php [hunk 4]
```diff
@@ -70,7 +71,7 @@ public function executeScripts()
             // Fallback to traditional handling for compatibility
             empty($this->installerScripts)
             // But not if typo3/cms is root package or typo3/cms is not found at all
-            && null !== $composer->getRepositoryManager()->getLocalRepository()->findPackage('typo3/cms', new EmptyConstraint())
+            && null !== $composer->getRepositoryManager()->getLocalRepository()->findPackage('typo3/cms', class_exists(EmptyConstraint::class) ? new EmptyConstraint() : new MatchAllConstraint())
```
Comment (helhum) [inline src/Plugin/Core/ScriptDispatcher.php:74]: `'*'`

**Review by helhum — CHANGES_REQUESTED** [review]

File: src/Plugin/Core/ScriptDispatcher.php (hunk 4 above)
Comment (benjaminkott) [reply src/Plugin/Core/ScriptDispatcher.php]: adapted

File: src/Plugin/Core/InstallerScripts/WebDirectory.php (hunk 3 above)
Comment (benjaminkott) [reply src/Plugin/Core/InstallerScripts/WebDirectory.php]: adapted

File: src/Plugin/Core/InstallerScripts/AutoloadConnector.php (hunk 2 above)
Comment (benjaminkott) [reply src/Plugin/Core/InstallerScripts/AutoloadConnector.php:60]: adapted

File: src/Plugin/Core/InstallerScripts/AutoloadConnector.php (hunk 1 above)
Comment (benjaminkott) [reply src/Plugin/Core/InstallerScripts/AutoloadConnector.php]: n1, adapted it

**Review by helhum — APPROVED** [review]

**Comment by benjaminkott** [comment]
@christian-ungur-at-trumpf-com 
hey, I am super sorry I missed those ones, also your mention got lost in GitHub email mess at my side - gladly @helhum hit me up on slack also thx again for the hint.

Added patches and tests for:
1.5.x branch https://github.com/TYPO3/CmsComposerInstallers/pull/110
1.3.x branch https://github.com/TYPO3/CmsComposerInstallers/pull/109

These should resolve your issue.

**Comment by helhum** [comment]
Released as 1.3.6 and 1.5.9

**Pull request merged**